### PR TITLE
P3 update 021723

### PR DESCRIPTION
**Pull request by hmorrison100**
TYPE: Bug fix + enhancement

KEYWORDS: P3, microphysics

SOURCE: Hugh Morrison (NCAR)

DESCRIPTION OF CHANGES: 

Minor bugs in the change in rain number during ice melting when ice amounts are tiny (almost no impact), 3-moment category interaction and merging (non-answer-changing in WRF), the GEM wrapper (not relevant to WRF), and a minor bug in lookup table 2 (formula for diagnostic mu_i). Inaccurate description of the output qzi variable in the Registry (non-answer-changing). Streamline and code cleanup needed for the lookup table generation codes.

 - P3 module code cleanup (e.g., consistent line spacing, clarifying comments, etc.)
 - Change in the internal P3 module name (thus requiring changes to the "use" statements in module_microphysics_driver.F and module_physics_init.F)
 - Minor bug fixes (addressing bugs noted above)
 - Refactoring of the GEM wrapper
 - Feature change to the lookup table 1 for 3-moment ice (changing number of mu indices from 11 to 10)
 - Cleanup of the lookup table generation codes
 - Fix to description of the output variable "qzi" in Registry.EM_COMMON
 - Link to new lookup table file names in the top-level Makefile

LIST OF MODIFIED FILES: 
	modified:   Makefile
	modified:   phys/module_microphysics_driver.F
	modified:   phys/module_mp_p3.F
	modified:   phys/module_physics_init.F
	deleted:    run/create_p3_lookupTable_1-2mom_v5.3.f90
	deleted:    run/create_p3_lookupTable_1-3mom_v5.1.7.f90
	deleted:    run/create_p3_lookupTable_2-v5.2.3.f90
	deleted:    run/p3_lookupTable_1.dat-3momI_v5.1.7
	deleted:    run/p3_lookupTable_1.dat-5.3-2momI
	deleted:    run/p3_lookupTable_2.dat-2momI_v5.2.3
New files added:
	run/create_p3_lookupTable_1.f90-v5.4
	run/create_p3_lookupTable_2.f90-v5.3
	run/p3_lookupTable_1.dat-v5.4_2momI
	run/p3_lookupTable_1.dat-v5.4_3momI
	run/p3_lookupTable_2.dat-v5.3

TESTS CONDUCTED:
1. All problems and issues have been resolved. P3v4.5.2 has been extensively tested in the GEM model for a large suite of real weather test cases. All four P3 mp options in WRF run successfully and give good results for the idealized supercell test case (with only minor changes form the previous version of P3).
2. Passed regression tests.

RELEASE NOTE: Further enhancements and minor bug fixes to P3 microphysics (updated to P3v4.5.2). Fix minor bugs in the change in rain number during ice melting when ice amounts are tiny (almost no impact), and a minor bug in lookup table 2 (formula for diagnostic mu_i). Update 3-moment category interaction and merging (non-answer-changing in WRF), and refactor the GEM wrapper (not relevant to WRF). Correct an inaccurate description of the output qzi variable in the Registry (non-answer-changing). Streamline and code cleanup for the lookup table generation codes.

### CODE REVIEW

**Comment by hmorrison100** [comment]
@mgduda @dudhia @weiwangncar I redid this to address the previous problems and created a new pull request. I created a new branch in my fork off of release-4.5, modified the files (so the commit shows the WRF files as modified rather than deleted and replaced to address @weiwangncar 's concern before). Let me know if there are any other problems. Thanks.

File: phys/module_physics_init.F [hunk 1]
```diff
@@ -4418,7 +4418,7 @@ SUBROUTINE mp_init(RAINNC,SNOWNC,GRAUPELNC,config_flags,restart,warm_rain,
    USE module_mp_fast_sbm
 #endif
    USE module_mp_morr_two_moment
-   USE module_mp_p3
+   USE microphy_p3
```
Comment (weiwangncar) [inline phys/module_physics_init.F:4421]: @dudhia Are you ok with the module name change inside the code?

**Comment by weiwangncar** [comment]
The Jenkins test results:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

File: phys/module_physics_init.F (hunk 1 above)
Comment (dudhia) [reply phys/module_physics_init.F:4421]: Yes, I think we can live with it. They want to share it with another model.

**Review by weiwangncar — APPROVED** [review]

**Review by dudhia — APPROVED** [review]

**Comment by weiwangncar** [comment]
@hmorrison100 I've copied and modified your text to RELEASE NOTES in the PR. Can you take a look and make necessary changes if needed? This bit is what's going to be included in our release announcement, hence some details would be needed. Thanks!

**Comment by hmorrison100** [comment]
@weiwangncar The modified release note looks great, thanks Wei!

**Pull request merged**